### PR TITLE
Add pull request template and release notes

### PR DESCRIPTION
**Pull request by nick-harder**
-adds a pull request template for future commits
-adds release notes to the documentation

### CODE REVIEW

**Review by maurerle — COMMENTED** [review]
Why is it needed to have a separate release_notes.rst instead of linking to `https://github.com/assume-framework/assume/releases` ?

**Comment by nick-harder** [comment]
> Why is it needed to have a separate release_notes.rst instead of linking to `https://github.com/assume-framework/assume/releases` ?

it is a little bit difficult to maintain the changelog and have the info for the releases. It would be best if people can maintain it with each commit and we can copy paster it during next release

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+- **Adjust the Framework to Schedule Storing to the Learning Role:** This enhancement enables Learning agents to participate in sequential
+  markets, such as day-ahead and intraday markets. The rewards are now written after the last market, ensuring that the learning process
+  accurately reflects the outcomes of all market interactions. This improvement supports more sophisticated and realistic agent training scenarios.
+  A tutorial on how to use this feature is coming soon.
+
+**Improvements:**
+
+- Significant speed up of the framework and especially of the learning process
+- Split scenario loader function to improve speed and reduce unrequired operations
+- Refactored unit operator by adding a seperate unit operator for learning units
```
Comment (maurerle) [inline docs/source/release_notes.rst:27]: separate

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Please check all applicable items:
+
+- [ ] Code changes are sufficiently documented (docstrings, inline comments, `doc` folder updates)
+- [ ] New unit tests added for new features or bug fixes
+- [ ] Existing tests pass with the changes
+- [ ] Reinforcement learning examples are operational (for DRL-related changes)
+- [ ] Code tested with both local and Docker databases
+- [ ] Code follows project style guidelines and best practices
+- [ ] Changes are backwards compatible, or deprecation notices added
+- [ ] New dependencies added to `environment.yaml`, `environment_docs.yaml`, and `setup.py`
```
Comment (maurerle) [inline .github/pull_request_template.md:33]: This is not applicable - only to `pyproject.toml`

**Review by maurerle — APPROVED** [review]
Sounds good. I would wait for Kim's feedback

File: docs/source/release_notes.rst (hunk 1 above)
Comment (maurerle) [reply docs/source/release_notes.rst:27]: Sorry, I meant that there is a typo: `separate` instead of `seperate :D
though your change is also fine :D

**Review by kim-mskw — APPROVED** [review]

**Pull request merged**